create_shards.c

```diff
@@ -91,12 +103,69 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 		ereport(ERROR, (errmsg("could not find column: %s", partitionColumnName)));
 	}
 
-	/* we only support hash partitioning method for now */
-	if (partitionMethod != HASH_PARTITION_TYPE)
+	partitionColumnVar = ColumnNameToColumn(distributedTableId, partitionColumnName);
+	partitionColumnTypeId = partitionColumnVar->vartype;
+
+	if (partitionMethod == HASH_PARTITION_TYPE)
+	{
+		partitionColumnOpClassId = GetDefaultOpClass(partitionColumnTypeId, HASH_AM_OID);
+	}
+	else
 	{
+		/* we only support hash partitioning right now */
 		ereport(ERROR, (errmsg("unsupported partition method: %c", partitionMethod)));
```

Comment:
Since you're modifying this function, can you update this `ereport` call to use the `ERRCODE_FEATURE_NOT_SUPPORTED` status code? It's most appropriate here.

I like the error message exactly as it stands, though we should add an error detail: _pg_shard only supports hash partitioning._